src/util.c

```diff
@@ -51,6 +51,10 @@
 #include "sha256.h"
 #include "config.h"
 
+#ifdef USE_FAST_FLOAT
+#include "../deps/fast_float/fast_float_strtod.h"
+#endif
```

Comment:
```suggestion
#ifdef USE_FAST_FLOAT
#include "../deps/fast_float/fast_float_strtod.h"
static inline double valkey_strtod(const char *str, char **endptr) {
    errno = 0;
    return fast_float_strtod(str, endptr);
}
#else
static inline double valkey_strtod(const char *str, char **endptr) {
    return strtod(str, endptr);
}
#endif
```
This cuts down the usage of ifdefs to a single location in this file. This can even be just in the header if we really want.

Reply:
Incorporated this into the new structure. Changed the interface though.